### PR TITLE
Add date nanoseconds field type

### LINKED ISSUE #4090: [DOC] supported field type `date_nanos`
The [list of supported fields types](https://opensearch.org/docs/latest/field-types/supported-field-types/index/) do not contain all supported field types available in OpenSearch.

One specific field type is `date_nanos`, which is supported in [OpenSearch](https://github.com/opensearch-project/OpenSearch/blob/af78ef9ea205917c56e9a3049e0b6d801f6efafd/server/src/main/java/org/opensearch/index/mapper/DateFieldMapper.java#LL93). The `date` type defaults stores dates in milliseconds where `date_nanos` stores in nanoseconds when no format is defined. If the format on the field is `strict` then users who attempt to store nanoseconds into milliseconds will be rejected.

This also impacts usage in OpenSearch Dashboards as it will format data and labels in visualizations based on the date format defined in the index pattern which default pulls it's type from OpenSearch. So if the index defined a field as `date` but create an index pattern using the types defined from index. Their data might have timestamps that are not precise to what is available. [[MORE INFO]](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3600#issuecomment-1544401557).

** Requirement **

Update supported field types to mention `date_nanos`

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #4090 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@kavilla Could you please review this PR for technical accuracy? Thanks!

File: _field-types/supported-field-types/date-nanos.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+Queries on `date_nanos` fields are converted to range queries on the field value's `long` representation. Then the stored fields and aggregation results are converted to a string using the format set on the field. 
+
+The `date_nanos` field supports all [formats]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/date#formats) and [parameters]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/date#parameters) that `date` supports. You can use multiple formats separated by `||`.
+{: .note}
+
+For `date_nanos` fields, you can use the `strict_date_optional_time_nanos` format to preserve nanosecond resolution. If you don't specify the format when mapping a field as `date_nanos`, the default format is `strict_date_optional_time||epoch_millis` that lets you pass values in either `strict_date_optional_time` or `epoch_millis` format. The `strict_date_optional_time` format supports dates in nanosecond resolution but the `epoch_millis` format supports dates in millisecond resolution only.
```
Comment (vagimeli) [inline _field-types/supported-field-types/date-nanos.md:19]:
```suggestion
For `date_nanos` fields, you can use the `strict_date_optional_time_nanos` format to preserve nanosecond resolution. If you don't specify the format when mapping a field as `date_nanos`, the default format is `strict_date_optional_time||epoch_millis` that lets you pass values in either `strict_date_optional_time` or `epoch_millis` format. The `strict_date_optional_time` format supports dates in nanosecond resolution, but the `epoch_millis` format supports dates in millisecond resolution only.
```

File: _field-types/supported-field-types/string.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ redirect_from:
 
 # String field types
 
-The following table lists all string field types that OpenSearch supports.
+String field types contains text values or values derived from text. The following table lists all string field types that OpenSearch supports.
```
Comment (vagimeli) [inline _field-types/supported-field-types/string.md:15]:
```suggestion
String field types contain text values or values derived from text. The following table lists all string field types that OpenSearch supports.
```

**Review by vagimeli — APPROVED** [review]
Minimal edits. Ready to merge.

**Pull request merged**